plugins/serializers/binary/binary.go

```diff
@@ -0,0 +1,133 @@
+package binary
+
+import (
+	"encoding/binary"
+	"fmt"
+	"log"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/plugins/serializers"
+)
+
+type Serializer struct {
+	Entries    []Entry `toml:"entries"`
+	Endianness string  `toml:"endianness"`
+
+	converter binary.ByteOrder
+}
+
+func (s *Serializer) Init() error {
+	log.Printf("D! [binary] binary serializer init")
+
+	if s.Endianness == "" {
+		s.Endianness = "host"
+	}
+
+	switch s.Endianness {
+	case "big":
+		s.converter = binary.BigEndian
+	case "little":
+		s.converter = binary.LittleEndian
+	case "host":
+		s.converter = internal.HostEndianness
+	default:
+		return fmt.Errorf("invalid endianness %q", s.Endianness)
+	}
+
+	for i, entry := range s.Entries {
+		if err := entry.Check(); err != nil {
+			log.Printf("E! [binary] entry: %s check failed: %v\n", entry.Name, err)
+			return err
+		}
+
+		s.Entries[i] = entry
+	}
+
+	return nil
+}
+
+func (s *Serializer) Serialize(metric telegraf.Metric) ([]byte, error) {
+	serialized := make([]byte, 0)
+
+	for _, entry := range s.Entries {
+		log.Printf("D! [binary] entry: %v\n", entry.Name)
+
+		switch entry.ReadFrom {
+		case "field":
+			field, found := metric.GetField(entry.Name)
+			log.Printf("D! [binary] field: %v, [%T]\n", field, field)
+			if !found {
+				return nil, fmt.Errorf("field %s not found", entry.Name)
+			}
+
+			entryBytes, err := entry.SerializeValue(field, s.converter)
+			log.Printf("D! [binary] field entry bytes: %x\n", entryBytes)
```

Comment:
I'm not sure about that. When user tweaks a receiver I feel it can be actually very helpful to see what data is serialized, how is it encoded to understand what party should be fixed. For me when preparing demo that is showed at the end of readme it was helpful

Reply:
Resolved, debug logs removed..